project_structure.txt

```diff
@@ -0,0 +1,119 @@
+.modin/
+   .config/
+   .distributed/
+   .pandas/
+   .test/
+   .rest files
+   .core/
+      .io/
+         .column_stores/
+            . # the same as it is now
+         .sql/
+            . # the same as it is now
+         .text/
+            . # the same as it is now
+         .file_dispatcher.py # the same as it is now
+         .io.py # the same as it is now
+      .dispatching/
+         .factories/
+            . # the same as it is now
+      .formats/
+         .base/
+            .query_compiler.py # BaseQueryCompiler
+         .pandas/
+            .base/ # or default (DefaultPandasQueryCompiler)
+               .query_compiler.py # BasePandasQueryCompiler
+            .kernels.py
+            .query_compiler.py # PandasQueryCompiler
+            .parsers.py
+            .utils.py # data_management/utils moved here
+      .execution/
```

Comment:
Should this go in `dataframe` folder?

Reply:
`dataframe` folder would look like this:
``` python
.dataframe/
   .algebra/
      .operators/
         . # the same as it is in functions
   .implementations/ # or `execution` would be better I think
      .base/
         .dataframe/
            .dataframe.py # BaseDataframe
            .metadata.py # BaseDataframeMetaData
         .partitioning/
            .axis_partition.py # BaseDataframeAxisPartition
            .partition_manager.py # BaseDataframeManager
            .partition.py # BaseDataframePartition
      .pandas/
         .dataframe/
            .dataframe.py # PandasDataframe
            .metadata.py # PandasDataframeMetaData
         .partitioning/
            .axis_partition.py # PandasDataframeAxisPartition
            .partition_manager.py # PandasDataframeManager
            .partition.py # PandasDataframePartition
      .ray/
         .generic/
            # GenericRayDataframe and related stuff
         .pandas_on_ray/
            .dataframe/
               .dataframe.py # PandasOnRayDataframe
               .metadata.py # PandasOnRayDataframeMetadata
            .partition_management/ # or `partitioning`
               .axis_partition.py # PandasOnRayDataframeAxisPartition
               .partition_manager.py # PandasOnRayDataframeManager
               .partition.py # PandasOnRayDataframePartition
            .io/
               .io.py # PandasOnRayIO
         .pandas_on_dask/
            . # the same as for pandas_on_ray
         .pandas_on_python/
            . # the same as for pandas_on_ray
```

Reply:
But in that case we lose `execution` word that I'd like we have since we have an utility function to get current execution (backend).
https://github.com/modin-project/modin/blob/57bcfc130e9fbe93d41aa6403b80782ef49eb9db/modin/utils.py#L509

Reply:
maybe we should rename `implementations` to `execution`. What do you think?